### DISCUSSION TITLE
Close CommandBar Overflow/Flyout after item has been selected?

### DISCUSSION THREAD

**Discussion by DanielLukic** [bdlukaa/fluent_ui#1061] [Q&A] [answered]
I must be missing something obvious. But how would I close the dynamic overflow of a `CommandBar` when an overflow item is selected by the user? Can I somehow tell from within the `onPressed` if a) the overflow is open and/or this is an overflow item? and b) access the overflow somehow to close it? or c) set some flag to do this automatically? Sorry for asking a n00b question here. Was looking at the source a bit. But not sure where to look tbh.. :shrug: :see_no_evil: 

My current workaround is to always call `pop` on the `Navigator` from every `CommandBarItem.onPressed`. Both primary and secondary. Because I'm not sure how to differentiate. Well, this works for me because I have no pages/routes/anything. But this can't be it, can it? :upside_down_face: 

Again, I must be overlooking something. Maybe someone can point me in the right direction? Thank you kindly!

**Comment by DanielLukic** [comment]
I just looked at the code I copied from the examples once more. Doh. `CommandBarBuilderItem` this will provide `CommandBarBuilderItem` to the builder function. This solves part of the problem. New to flutter. Still learning the basics, I guess.. :see_no_evil:

**Reply by DanielLukic** [reply]
Just a quick hack to show what I have so far:
```dart
extension on BuildContext {
  CommandBarBuilderItem _buildItem(
    String label,
    String tooltip,
    IconData icon,
    void Function() onPressed,
  ) {
    var inSecondary = false;
    void closeIfSecondary() {
      if (!inSecondary) return;
      Navigator.of(this).maybePop();
    }

    return CommandBarBuilderItem(
      builder: (context, mode, w) {
        inSecondary = mode == CommandBarItemDisplayMode.inSecondary;
        return Tooltip(message: tooltip, child: w);
      },
      wrappedItem: CommandBarButton(
        icon: Icon(icon),
        label: Text(label),
        onPressed: () {
          onPressed();
          closeIfSecondary();
        },
      ),
    );
  }
}
```
:see_no_evil:

**Comment by bdlukaa** [comment, marked as answer]
The flyout is displayed as a route on the app, so you can call `Navigator.pop` to close it. It works for all fluent_ui Flyouts.
